crates/sui-indexer-alt/src/lib.rs

```diff
@@ -415,6 +416,11 @@ pub async fn start_indexer(
         max_chunk_size: 5 * 300,
     });
 
+    ensure!(
+        extra.is_empty(),
+        "Unexpected pipeline configurations (maybe a typo?):\n{extra}",
+    );
+
```

Comment:
I think this is a nice middle ground. If this was a CLI tool vs a service we'd probably want to "warn" vs error out but I think this is the correct approach for this service.

Reply:
In the end, I think I made this a warning for the forwards compatibility reasons that you mentioned (in the merging configs PR).